lib/collection/src/operations/verification/mod.rs

```diff
@@ -251,3 +251,184 @@ impl StrictModeVerification for SearchParams {
         None
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::sync::Arc;
+
+    use common::cpu::CpuBudget;
+    use segment::types::{
+        Condition, FieldCondition, Filter, Match, SearchParams, StrictModeConfig,
+    };
+    use tempfile::Builder;
+
+    use super::StrictModeVerification;
+    use crate::collection::{Collection, RequestShardTransfer};
+    use crate::config::{CollectionConfig, CollectionParams, WalConfig};
+    use crate::operations::point_ops::{FilterSelector, PointsSelector};
+    use crate::operations::shared_storage_config::SharedStorageConfig;
+    use crate::operations::types::{
+        CollectionError, CountRequestInternal, DiscoverRequestInternal,
+    };
+    use crate::optimizers_builder::OptimizersConfig;
+    use crate::shards::channel_service::ChannelService;
+    use crate::shards::collection_shard_distribution::CollectionShardDistribution;
+    use crate::shards::replica_set::{AbortShardTransfer, ChangePeerState};
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_strict_mode_verification_trait() {
+        let collection = fixture().await;
+
+        test_query_limit(&collection).await;
```

Comment:
The timeout function seems to be obsolete, and completely replaced by the timeout parameter in `check_strict_mode`. Therefore there is no test for timeout here.

I've kept the function in the trait, since we might need it in future.
If you think we should rather remove it and add it back in case we need it, I can do that in a following PR.

Reply:
> If you think we should rather remove it and add it back in case we need it, I can do that in a following PR.

If we don't use it, let's remove it. We can always put it back later.

Just to be clear, we do validate it as part of strict mode, but at the endpoint level. Not on the internal structures through the trait.

Reply:
> If you think we should rather remove it and add it back in case we need it, I can do that in a following PR.

If we don't use it, let's remove it. We can always put it back later.

Just to be clear, we do validate it as part of strict mode, but at the endpoint level. Not on the internal structures through the trait.

Reply:
I'll do this in a following PR since there is no overlap with unit tests.